substrate/pointer_utils

```diff
@@ -0,0 +1,117 @@
+// SPDX-License-Identifier: BSD-3-Clause
+#ifndef SUBSTRATE_POINTER_UTILS
+#define SUBSTRATE_POINTER_UTILS
+
+#include <functional>
+#include <memory>
+
+#include "internal/defs"
+#include "utility"
+
+#if __cplusplus >= 201304L
+#define SUBSTRATE_SUPPORTS_MANAGED_TUPLE
+
+namespace substrate
+{
+	template<typename, typename...> class managedTupleWithValues_t;
+	template<typename, typename...> class managedTuple_t;
+
+	template<typename R, typename... Args, typename... Vals> class managedTupleWithValues_t<R(Args...), Vals...>
+	{
+		static_assert(sizeof...(Args) == sizeof...(Vals), "Mismatching arguments and variable types! Did you override them all?");
+		static_assert(sizeof...(Args) > 0, "You can't run a deleter with no arguments");
+		static_assert(sizeof...(Vals) > 0, "You can't run a deleter against no variables");
+
+	public:
+		managedTupleWithValues_t(std::function<R(Args...)> releaser)
+			: function(std::move(releaser))
+		{
+		}
+
+		managedTupleWithValues_t(managedTupleWithValues_t&) = delete;
+		managedTupleWithValues_t operator=(managedTupleWithValues_t&) = delete;
+		managedTupleWithValues_t(managedTupleWithValues_t&&) noexcept = default;
+		managedTupleWithValues_t &operator=(managedTupleWithValues_t&&) noexcept = default;
+
+		template<size_t I>
+		inline auto& get() noexcept
```

Comment:
Same line as the template keyword please. Use `auto &` rather than `auto& `

The function body can be collapsed to a single line indented from the function declaration as in:
```
template<size_t I> inline auto &get() noexcept
    { return std::get<I>(elements); }
```